compiler/testData/codegen/box/coroutines/featureIntersection/funInterface/kt50950.kt

```diff
@@ -0,0 +1,21 @@
+// IGNORE_BACKEND: JVM
+// WITH_COROUTINES
+// WITH_STDLIB
+// FILE: a.kt
+fun interface A {
+    suspend fun run()
+}
+
+// FILE: b.kt
+import kotlin.coroutines.*
+import kotlin.coroutines.intrinsics.*
+import helpers.EmptyContinuation
+
+suspend fun run(block: () -> Unit) = A(block).run()
+
+fun box(): String {
+    suspend {
+        run {}
+    }.startCoroutine(EmptyContinuation)
+    return "OK"
+}
```

Comment:
The test fails in FIR:
```
  java.lang.ClassCastException: BKt$box$1$1 cannot be cast to kotlin.jvm.functions.Function1
    at BKt.run(b.kt:14)
    at BKt$box$1.invokeSuspend(b.kt:18)
    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
    at kotlin.coroutines.ContinuationKt.startCoroutine(Continuation.kt:115)
    at BKt.box(b.kt:19)
```

Reply:
Fixed, FIR does not support using `() -> Unit` as `suspend () -> Unit` yet.